### ISSUE TITLE
new install instructions / Vagrant for Ubuntu 20.04 LTS

### ISSUE THREAD

**Issue by mtmail** [osm-search/Nominatim#1768]
Ubuntu 20.04 (Focal Fossa) got released https://wiki.ubuntu.com/FocalFossa/ReleaseNotes

Create install instructions similar to
https://github.com/osm-search/Nominatim/tree/master/vagrant
https://github.com/osm-search/Nominatim/blob/master/Vagrantfile

https://ubuntu.com/about/release-cycle

First to see what the new default versions of PHP, Python, Postgres are and if the various other dependencies changes, e.g. are they already available, did their name change?

Decision to point the main documention from 18 to 20, depreciate 16, or switching Travis CI from 16 to 18 or 20 will be made separately.

**Comment by Simon-Will** [comment]
Hello, I would like to contribute to Nominatim and this seems like a good way to get to know the project. So I’m going to work on this issue now, if it’s alright. I’ll report back later.

**Comment by Simon-Will** [comment]
I made it possible to run the Vagrantfile with Ubuntu 20.04. The instructions in [`VAGRANT.md`](https://github.com/osm-search/Nominatim/blob/42c80893cb1038e21c05c2d0d30e3a7a20f0fa26/VAGRANT.md) still work as before. I updated the names of the Vagrant machines so that Ubuntu 18.04 (previously called `ubuntu`) is now called `ubuntu18` and Ubuntu 20.04 is now called `ubuntu20`.

## Changes

The version changes from Ubuntu 18.04 to Ubuntu 20.04 are:

  - Python: 3.6 to 3.8
  - Postgres: 10 to 12
  - PHP: 7.2 to 7.4

In the `apt-get`, I changed `--force` to `--allow-downgrades --allow-remove-essential --allow-change-held-packages`, because the former is deprecated. Cf. the [manpage of apt-get](http://manpages.ubuntu.com/manpages/focal/man8/apt-get.8.html)

The php module `codesniffer` was previously installed via Composer, but it is available via the Ubuntu repository, so I installed it via `apt-get` now.

## Warnings during build

There were two warnings that were present before and are still present now:

### `pyosmium`

```
ubuntu20: CMake Warning at CMakeLists.txt:69 (message):                               
ubuntu20:   pyosmium-get-changes not found (required for updates)
```

Should we just install `python3-pyosmium`? Or is there a reason that it is not installed?

### Something with `install` target

```
ubuntu20: /home/vagrant/Nominatim/module/Makefile:11: warning: overriding recipe for target 'install'
ubuntu20: /usr/lib/postgresql/12/lib/pgxs/src/makefiles/pgxs.mk:241: warning: ignoring old recipe for target 'install
```

I don’t really understand where this comes from.

## Tests

### PHP

`phpunit` does not work any more. It fails with the following error:

```
PHP Fatal error:  Declaration of Nominatim\AddressDetailsTest::setUp() must be compatible with PHPUnit\Framework\TestCase::setUp(): void in /home/vagrant/Nominatim/test/php/Nominatim/AddressDetailsTest.php on line 12
```

This is because of updates to PHPUnit (6 to 8 between Ubuntu releases). I’m going to open a separate issue for this.

`phpcs` runs and exits successfully.

### Python

`behave -DBUILDDIR=/home/vagrant/build/ db osm2pgsql` runs successfully and produces the following output:

```
21 features passed, 0 failed, 0 skipped
149 scenarios passed, 0 failed, 1 skipped
865 steps passed, 0 failed, 9 skipped, 0 undefined
Took 5m13.278s
```

**Comment by Simon-Will** [comment]
I submitted pull request #1780 for this.

**Comment by mtmail** [comment]
Excellent, much less changes than expected. Ubuntu 16 in the last year was tricky to keep running as more and more packages depended on newer versions. We need to review if we want the recommeded installation to use Postgresql 12 already (https://github.com/osm-search/Nominatim/issues/1677)

> pyosmium-get-changes not found (required for updates)

That happens during every installation. In the installation instructions we first want users to install and import their data, and then later (optional) setup automatic data updates. "required for updates" is true, but it's not required for importig data.

**Comment by lonvia** [comment]
This has been implemented in #1780. Thanks @Simon-Will.
